### PR TITLE
chore: remove fasionmnist dependency from tests

### PR DESCRIPTION
**Pull request by fd0r**

### CODE REVIEW

File: tests/parameter_search/test_p_error_binary_search.py [hunk 1]
```diff
@@ -33,17 +31,17 @@
...
-        "train_transform": transforms.Compose(
-            [transforms.ToTensor(), transforms.Normalize(0.2859, 0.3530)]
-        ),
-        "test_transform": transforms.Compose(
-            [transforms.ToTensor(), transforms.Normalize(0.2859, 0.3530)]
-        ),
-        "dataset_size": 60000,
-        "n_classes": 10,
-    },
+    # "FashionMNIST": {
```
Comment (andrei-stoian-zama) [inline tests/parameter_search/test_p_error_binary_search.py:34]: why not remove it entirely ?

Comment (fd0r) [reply tests/parameter_search/test_p_error_binary_search.py:34]: Because it's the one causing all the flakies

Comment (fd0r) [reply tests/parameter_search/test_p_error_binary_search.py:34]: Looks like I'm missing coverage, I'll continue working on it on monday

Comment (fd0r) [reply tests/parameter_search/test_p_error_binary_search.py:34]: Ho you mean why I commented it, it's because it's a draft, I'll remove in the final PR

File: tests/parameter_search/test_p_error_binary_search.py [hunk 2]
```diff
@@ -426,6 +452,7 @@ def test_invalid_estimator_for_built_in_models(model_class, parameters, load_dat
         search.run(x=x_calib, ground_truth=y)
 
 
+# TODO: change from binary_classification_metric
```
Comment (RomanBredehoft) [inline tests/parameter_search/test_p_error_binary_search.py:455]: maybe add an issue here ?

File: tests/parameter_search/test_p_error_binary_search.py (hunk 1 above)
Comment (RomanBredehoft) [reply tests/parameter_search/test_p_error_binary_search.py:41]: you still want to remove it right ?

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
thanks ! a few changes are needed I think 

also, this should solve https://github.com/zama-ai/concrete-ml-internal/issues/3812 right ? if so, maybe also remove the flaky mark + the ignore in makefile (look for 3812 in the code)

**Comment by fd0r** [comment]
Ho yes nice catch indeed thanks, I'll clean the PR

**Comment by fd0r** [comment]
I removed commented code and the flaky marker on hybrid tests.
Also squashed everything in one commit.

File: tests/parameter_search/test_p_error_binary_search.py [hunk 3]
```diff
@@ -105,9 +84,22 @@ def binary_classification_metric(y_true: numpy.ndarray, y_pred: numpy.ndarray) -
...
+        float: Accuracy.
+    """
+    y_pred = y_prob.argmax(1)
+    return balanced_accuracy_score(y_true, y_pred)
+
+
 @pytest.mark.parametrize("attr, value", [("undefined_attr", 1)])
 @pytest.mark.parametrize(
-    "model_name, quant_type, metric", [("CustomModel", "qat", binary_classification_metric)]
+    "model_name, quant_type, metric", [("CustomModel", "qat", balanced_accuracy_score_argmax)]
```
Comment (RomanBredehoft) [inline tests/parameter_search/test_p_error_binary_search.py:102]: maybe add a comment somewhere on why we now chose this metric instead of the old one !

File: tests/torch/test_hybrid_converter.py [hunk 4]
```diff
@@ -55,7 +55,9 @@ def run_hybrid_model_test(
     accuracy = is_in.float().mean()
 
     # Make sure accuracy is above a certain threshold
-    assert accuracy >= expected_accuracy, "Expected accuracy GPT2 hybrid not matched."
+    # We need to add some tolerance otherwise the test is flaky
+    tolerance = 0.005
+    assert accuracy >= expected_accuracy - tolerance, "Expected accuracy GPT2 hybrid not matched."
```
[the remark this answers is not in the document]
Comment (fd0r) [reply tests/torch/test_hybrid_converter.py:60]: Would a single PR with 2 commits be okay with you?

I'll create an issue and add the link here

File: tests/parameter_search/test_p_error_binary_search.py [hunk 5]
```diff
@@ -371,21 +366,33 @@ def test_binary_search_for_built_in_models(model_class, parameters, threshold, p
...
+            keras.layers.Dense(
+                units=MODELS_ARGS[model_name]["dataset"]["n_classes"], activation="softmax"
+            ),
         ]
     )
 
-    dataset = get_torchvision_dataset(DATASETS_ARGS["FashionMNIST"], train_set=True)
-    x_calib, y = data_calibration_processing(dataset, n_sample=1)
+    # dataset = get_torchvision_dataset(DATASETS_ARGS["FashionMNIST"], train_set=True)
+    # x_calib, y = data_calibration_processing(dataset, n_sample=1)
```
Comment (kcelia) [inline tests/parameter_search/test_p_error_binary_search.py:387]: did you leave the comments in purpose or is it an oversight?

Comment (fd0r) [reply tests/parameter_search/test_p_error_binary_search.py:387]: An oversight, thanks for seeing it!

File: tests/torch/test_hybrid_converter.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (RomanBredehoft) [reply tests/torch/test_hybrid_converter.py:60]: I mean, it feels weird to tackle the hybrid flaky in a PR called "remove fashionmist" so yes I guess it would be better to have a small independent PR (better tracking as well) , but not a hug huge deal anyway

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
this change should resolve https://github.com/zama-ai/concrete-ml-internal/issues/3812 right ? if so, maybe remove the flaky mark and make the few changes in the makefile (make pytest_no_flaky)

File: Makefile [hunk 6]
```diff
@@ -250,8 +250,6 @@ pytest_macOS_for_GitHub:
 	--randomly-dont-reset-seed \
 	${PYTEST_OPTIONS}
 
-# Warning: collecting tests can trigger the parameter_search flaky test
-# FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3812
 .PHONY: check_currentl_flaky_tests # Print the current list of known flaky tests
```
Comment (RomanBredehoft) [inline Makefile:253]: just realized there's a small typo here `check_currentl_flaky_tests` should be `check_current_flaky_tests`

**Review by RomanBredehoft — APPROVED** [review]
Thanks a lot !
